lib/activity.js

```diff
@@ -36,79 +18,18 @@ class ActivityEvent {
     this.message = message
     this.id = id
   }
-
-  /**
-   * @param {String} line
-   * @returns {ActivityEvent}
-   */
-  static fromLogLine (line) {
-    const { timestamp, text } = parseLog(line)
-    const { type, message, source, id } = JSON.parse(text)
-    return new ActivityEvent({
-      timestamp,
-      type,
-      source,
-      message,
-      id
-    })
-  }
-}
-
-class Activity {
-  constructor (activityPath, logs) {
-    this.activityPath = activityPath
-    this.logs = logs
-  }
-
-  /**
-   * @returns {Promise<ActivityEvent[]>}
-   */
-  async get () {
-    return (await fs.readFile(this.activityPath, 'utf-8'))
-      .trim()
-      .split('\n')
-      .filter(Boolean)
-      .map(line => ActivityEvent.fromLogLine(line))
-  }
-
-  /**
-   * @param {Object} [options]
-   * @param {AbortSignal} [options.signal]
-   * @param {number} [options.nLines=10]
-   * @returns {AsyncGenerator<ActivityEvent>}
-   */
-  async * follow ({ signal, nLines = 10 } = {}) {
-    const tail = new Tail(this.activityPath, {
-      nLines,
-      useWatchFile: platform() === 'win32'
-    })
-    for await (const [line] of on(tail, 'line', { signal })) {
-      yield ActivityEvent.fromLogLine(line)
-    }
-  }
-
-  createWriteStream (source) {
-    const addMetaStream = new AddMetaStream({ source })
-    addMetaStream
-      .pipe(new SerializeStream())
-      .pipe(this.logs.createWriteStream(this.activityPath))
-    return addMetaStream
-  }
-
-  async maybeCreateActivityFile () {
-    await maybeCreateFile(this.activityPath)
-  }
 }
 
-const formatActivityObject = ({ type, message, timestamp }) => {
-  return formatLog(`${type.toUpperCase().padEnd(5)} ${message}`, {
-    timestamp,
-    pretty: true
-  })
+const formatActivityObject = ({ type, message }) => {
+  return `${type.toUpperCase().padEnd(5)} ${message}`
+    .trimEnd()
+    .split(/\n/g)
+    .map(line => `[${(new Date()).toLocaleString()}] ${line}`)
+    .join('\n') + '\n'
 }
 
 module.exports = {
+  activities: new EventEmitter(),
```

Comment:
I have mixed feelings about exposing a naked EventEmitter instance. It's super easy to implement, there is no unnecessary code. But: When reading the source code in this file, I have no idea what events will be emitted.

I prefer the approach in `lib/metrics.js`, where you export a class instance that provides methods for submitting new metrics data. That makes it much more clear how to work with the metrics module.

Would you mind using the same approach for activities too?

Instead of `submit`, maybe we can have `record` or `publish`? (Hmm, it's almost the same thing as `emit`, isn't it? 🤔)